### ISSUE TITLE
How to add code coverage for compile_native_go_fuzzer?

### ISSUE THREAD

**Issue by antonmedv** [google/oss-fuzz#10916]
I see what project expr has no coverage report.

From the documentation looks like go-fuzz should have cov reporting, but not compile_native_go_fuzzer, is that right?

How to add coverage for compile_native_go_fuzzer?

Thanks.

**Comment by DavidKorczynski** [comment]
@AdamKorcz could you assist here?

**Comment by antonmedv** [comment]
Seems what coverage for expr still failing. 🤔

**Comment by antonmedv** [comment]
Looks like code at https://github.com/google/oss-fuzz/blob/d4f443201aa7fef94f597e49ca768cfcabe9e25b/infra/base-images/base-builder/compile_native_go_fuzzer#L52C1-L52C1 selects all go files. In my expr there are two go files. So next cp fails.

**Comment by AdamKorcz** [comment]
Yes, it might be that it is the `-r` flag that causes issues. I assume you have two `fuzz_test.go`?

**Comment by antonmedv** [comment]
- [fuzz_test.go](https://github.com/antonmedv/expr/blob/master/test/fuzz/fuzz_test.go)
- [fuzz_coverage_test.go](https://github.com/antonmedv/expr/blob/master/test/fuzz/fuzz_coverage_test.go)

**Comment by antonmedv** [comment]
I can delete [fuzz_coverage_test.go](https://github.com/antonmedv/expr/blob/master/test/fuzz/fuzz_coverage_test.go)

**Comment by antonmedv** [comment]
After dropping fuzz_coverage_test file coverage reporting works.  But it strange: coverage reported only for one file: https://storage.googleapis.com/oss-fuzz-coverage/expr/reports/20230917/linux/index.html#file0

<img width="1465" alt="image" src="https://github.com/google/oss-fuzz/assets/141232/0b226641-022d-4001-a2fd-dbae3be33580">

**Comment by AdamKorcz** [comment]
> After dropping fuzz_coverage_test file coverage reporting works. But it strange: coverage reported only for one file: https://storage.googleapis.com/oss-fuzz-coverage/expr/reports/20230917/linux/index.html#file0
> 
> <img alt="image" width="1465" src="https://user-images.githubusercontent.com/141232/268652888-0b226641-022d-4001-a2fd-dbae3be33580.png">
> Only fuzz_env.go file in coverage.

Thank you for testing it out @antonmedv; I am happy that it is working. The issue with only showing a single file is in Golang: AFAIK, Golang only includes the packages that have coverage in the report.

Closing, since things are fixed on the OSS-Fuzz side.

**Comment by antonmedv** [comment]
But why Expr source files were not included in the coverage report? fuzz_text calls expr.Run() but it is not included.

**Comment by AdamKorcz** [comment]
> But why Expr source files were not included in the coverage report? fuzz_text calls expr.Run() but it is not included.

Let me check that.

**Comment by antonmedv** [comment]
Coverage is still reported only for one file (100%):

<img width="214" alt="Screenshot 2023-10-02 at 22 53 31" src="https://github.com/google/oss-fuzz/assets/141232/2959ced7-4068-4362-98e6-945d4c470720">
<img width="980" alt="Screenshot 2023-10-02 at 22 53 52" src="https://github.com/google/oss-fuzz/assets/141232/dfd888df-8608-4532-8e70-362801eb6261">
